app/models/profile.rb

```diff
@@ -70,5 +70,24 @@ def from_openscap_parser(op_profile, benchmark_id: nil, account_id: nil)
 
       profile
     end
+
+    def rules_and_rule_groups(profile)
+      profile_rules = profile.rules
+      profile_rule_groups = profile.rule_groups
+      custom_arrange(profile_rule_groups, profile_rules)
+    end
+
+    def custom_arrange(rule_groups, rules)
+      all_rule_children = rules.left_outer_joins(:rule_groups).where(rule_groups: { id: rule_groups.select(:id) })
+
+      arranged_rule_groups = rule_groups.arrange_serializable do |parent, children|
+        rule_children = parent.rules.where(rules: { id: rules.select(:id) }).as_json(root: :rule)
```

Comment:
These two together might be inefficient, not sure how `arrange_serializable` works, but you can check the logs for how many queries it generates for a larger profile.

Reply:
Yeah, that's true. It might be inefficient for profiles with a lot of rule_groups because `rule_children = parent.rules.where(rules: { id: rules.select(:id) }).as_json(root: :rule)` is executed for every rule_group that belongs to the profile and the query takes about `1.5ms` every time. Would this be too slow? I can check to see what the max number of rule_groups under a profile is. 
If this is too slow, I can come up with a different way of adding the rule children to the hash